extensions/AMD/SPV_AMDX_shader_enqueue.asciidoc

```diff
@@ -217,47 +270,106 @@ Modify Section 3.21, "BuiltIn", adding the following row to the table:
 [cols="1,10,8",options="header"]
 |====
 2+^.^| BuiltIn | Enabling Capabilities
+| 5021 | *RemainingRecursionLevelsAMDX* +
+The number of times this node can still enqueue payloads for itself. +
+Is equal to 0 if at the leaf or if the node is not recursive at all.
+|*ShaderEnqueueAMDX*
 | 5073 | *ShaderIndexAMDX* +
 Index assigned to the current shader.
 |*ShaderEnqueueAMDX*
-| 5021 | *CoalescedInputCountAMDX* +
-Number of valid inputs in the *NodePayloadAMDX* storage class array when using the *CoalescingAMDX* Execution Mode. (GLCompute only)
-|*ShaderEnqueueAMDX*
 |====
 
 == Instructions
 
 Add the following new instructions:
 
-[cols="1,2,2,2,2,2"]
+[cols="4*1"]
+|======
+3+|[[OpConstantStringAMDX]]*OpConstantStringAMDX* +
+ +
+Declare a new string specialization constant. +
+ +
+_String_ is the value of the constant. +
+ +
+Unlike *OpString*, this is a semantically meaningful instruction and cannot be safely removed from a module.
+1+|Capability: +
+*ShaderEnqueueAMDX*
+| 3 + variable | 5103
+| _Result <id>_
+| _Literal_ +
+_String_
+|======
+
+[cols="4*1"]
+|======
+3+|[[OpSpecConstantStringAMDX]]*OpSpecConstantStringAMDX* +
+ +
+Declare a new string specialization constant. +
+ +
+_String_ is the default value of the constant. +
+ +
+Unlike *OpString*, this is a semantically meaningful instruction and cannot be safely removed from a module. +
+ +
+This instruction can be specialized to become an *OpConstantStringAMDX* instruction. +
+ +
+See _Specialization_.
+1+|Capability: +
+*ShaderEnqueueAMDX*
+| 3 + variable | 5104
+| _Result <id>_
+| _Literal_ +
+_String_
 |======
-5+|[[OpInitializeNodePayloadsAMDX]]*OpInitializeNodePayloadsAMDX* +
+
+[cols="4*1",width="100%"]
+|=====
+3+|[[OpTypeNodePayloadArrayAMDX]]*OpTypeNodePayloadArrayAMDX* +
  +
-Allocate payloads in memory and make them accessible through the _Payload Array_ variable.
-The payloads are enqueued for the node shader identified by the _Node Index_ and _Node Name_ in the decoration
-*PayloadNodeNameAMDX* on the _Payload Array_ variable. +
+Declare a new payload array type.  Its length is not known at compile time. +
+ +
+_Payload Type_ is the type of each payload in the array. +
+ +
+ See <<OpNodePayloadArrayLengthAMDX,*OpNodePayloadArrayLengthAMDX*>> for getting the length of an array of this type. +
+ +
+A payload array can be allocated by either *OpAllocateNodePayloadsAMDX* to be enqueued as an output, or via *OpVariable* in the *NodePayloadAMDX* storage class to be consumed as an input. +
+ +
+Can be dereferenced using an access chain in the same way as *OpTypeRuntimeArray* or *OpTypeArray*.
+1+|<<Capability,Capability>>: +
+*Shader*
```

Comment:
https://github.com/KhronosGroup/SPIRV-Headers/pull/452 ties it to `ShaderEnqueueAMDX` not `Shader`

Reply:
Yes, thank you, both (this and the other comment) should be `ShaderEnqueueAMDX` as these are the new `*AMDX` instructions.